crates/aptos/src/update/prover_dependencies.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    cli_build_information,
+    common::types::{CliCommand, CliError, CliTypedResult},
+    update::{
+        get_additional_binaries_dir, prover_dependency_installer::DependencyInstaller,
+        update_binary,
+    },
+};
+use anyhow::{Context, Result};
+use aptos_build_info::BUILD_OS;
+use async_trait::async_trait;
+use clap::Parser;
+use move_prover_boogie_backend::options::{
+    BoogieOptions, MAX_BOOGIE_VERSION, MAX_CVC5_VERSION, MAX_Z3_VERSION, MIN_BOOGIE_VERSION,
+    MIN_CVC5_VERSION, MIN_Z3_VERSION,
+};
+use std::{
+    env,
+    path::{Path, PathBuf},
+};
+
+pub(crate) const REPO_NAME: &str = "prover-dependency";
+pub(crate) const REPO_OWNER: &str = "aptos-labs";
+
+pub(crate) const BOOGIE_BINARY_NAME: &str = "boogie";
+pub(crate) const TARGET_BOOGIE_VERSION: &str = "3.2.4";
+
+pub(crate) const BOOGIE_EXE_ENV: &str = "BOOGIE_EXE";
+#[cfg(target_os = "windows")]
+pub(crate) const BOOGIE_EXE: &str = "boogie.exe";
+#[cfg(not(target_os = "windows"))]
+pub(crate) const BOOGIE_EXE: &str = "boogie";
+
+const Z3_BINARY_NAME: &str = "z3";
+const TARGET_Z3_VERSION: &str = "4.11.2";
+
+const Z3_EXE_ENV: &str = "Z3_EXE";
+#[cfg(target_os = "windows")]
+const Z3_EXE: &str = "z3.exe";
+#[cfg(not(target_os = "windows"))]
+const Z3_EXE: &str = "z3";
+
+const CVC5_BINARY_NAME: &str = "cvc5";
+const TARGET_CVC5_VERSION: &str = "0.0.3";
+
+const CVC5_EXE_ENV: &str = "CVC5_EXE";
+#[cfg(target_os = "windows")]
+const CVC5_EXE: &str = "cvc5.exe";
+#[cfg(not(target_os = "windows"))]
+const CVC5_EXE: &str = "cvc5";
+
+/// Install dependencies (boogie, z3 and cvc5) for Move prover
+#[derive(Debug, Parser)]
+pub struct ProverDependencyInstaller {
+    /// Where to install binaries of boogie, z3 and cvc5. If not
+    /// given we will put it in a standard location for your OS.
+    #[clap(long)]
+    install_dir: Option<PathBuf>,
+}
+
+impl ProverDependencyInstaller {
+    fn add_env_var(&self, env_var: &str, install_path: &Path) -> Result<(), CliError> {
+        if let Ok(current_value) = env::var(env_var) {
+            if current_value == install_path.to_string_lossy() {
+                return Ok(());
+            } else {
+                return Err(CliError::UnexpectedError(format!(
+                    "{} is already set to a different value: {}.",
+                    env_var, current_value
+                )));
+            }
+        }
+
+        set_env::set(env_var, install_path.to_string_lossy())
+            .map_err(|e| CliError::UnexpectedError(format!("Failed to set {}: {}", env_var, e)))?;
+        println!(
+            "Added {} to environment with value: {}.",
```

Comment:
This message should make it clear that we actually added this to their profile, so it is added to the environment permanently.